src/_pytest/pastebin.py

```diff
@@ -67,8 +67,13 @@ def create_new_paste(contents):
 
     params = {"code": contents, "lexer": "python3", "expiry": "1week"}
     url = "https://bpaste.net"
-    response = urlopen(url, data=urlencode(params).encode("ascii")).read()
-    m = re.search(r'href="/raw/(\w+)"', response.decode("utf-8"))
+    try:
+        response = (
+            urlopen(url, data=urlencode(params).encode("ascii")).read().decode("utf-8")
+        )
+    except OSError as exc_info:  # urllib errors
```

Comment:
I don't think we should silently gobble the errors here, if someone requests `--pastebin` and pytest fails to pastebin then the command should probably exit nonzero

Reply:
Our use-case is a home-spun continuous integration script for https://www.qdyn-library.net. This project is hosted on a gitolite installation on a university workstation running commercial Fortran compilers, which also acts as the CI server via a simple post-commit hook kicking off a [SLURM](https://slurm.schedmd.com) job to run the test matrix. Everything is totally headless and not web-based, so the CI-server sends developers an email with the results of the CI. These only include a summary, the detailed test output tends to be a bit too large to send around by email, so the CI-server uses `--pastebin` to upload the output, and puts a link in the email.

Recently, this started failing for some test runs with an HTTP 400 error. We're not quite sure what exactly the cause is. In any case, having the full test output available is very much secondary to the actual tests passing or failing. We don't want to get a "test failure" just because the upload to pastebin failed for some reason. After all, it might just be a temporary network failure. It would be enough to have the appropriate error message in the email (as this PR implements). We can then still investigate what the problem is, but a successful upload to pastebin is no longer a necessary condition for accepting patches. So to "silently gobble the errors" is explicitly the point here. The error will still be logged to the output, so it's not *totally* silent. Of course, this behavior is debatable. It's just that with the current behavior of pytest failing if pastebin fails, we'd probably have to stop using that feature and re-implement it as a pytest-plugin that matches our use case; that would obviously be more work than this small patch.

Reply:
A possible compromise *might* be to exit with a specific status code that indicates "Everything OK except for the `--pastebin`". The CI-script might be able to catch that as a special case.

Reply:
I'm OK with catching and displaying the error instead of returning non-zero. I think most (all?) use cases for `--pastebin` follow the pattern of "posting to pastebin for a nice-to-have report" but that is not really required.

Either way, this should target `features` as it changes the behavior. 👍

Reply:
So should I rename `changelog/5764.improvement.rst` to `changelog/5764.feature.rst` and reword the changelog text?

Reply:
I changed the base branch from `master` to `features` just now.